perses/rjmc/topology_proposal.py

```diff
@@ -1470,6 +1474,98 @@ def generate_oemol_from_pdb_template(pdbfile):
             atom.SetName(name_without_spaces)
         return current_oemol
 
+    @staticmethod
+    def _get_charge_difference(current_resname, new_resname):
+        """
+        return the charge of the new res - charge old res
+
+        Parameters
+        ----------
+        current_resname : str
+            three letter identifier for original residue
+        new_resnme : str
+            three letter identifier for new residue
+
+        Returns
+        -------
+        chargediff : int
+            charge(new_res) - charge(old_res)
+        """
+        assert new_resname in PolymerProposalEngine._aminos
+        assert current_resname in PolymerProposalEngine._aminos
+
+        new_rescharge, current_rescharge = 0,0
+        resname_to_charge = {current_resname: 0, new_resname: 0}
+        for resname in [new_resname, current_resname]:
+            if resname in PolymerProposalEngine._negative_aminos:
+                resname_to_charge[resname] -= 1
+            elif resname in PolymerProposalEngine._positive_aminos:
+                resname_to_charge[resname] += 1
+
+        return resname_to_charge[new_resname] - resname_to_charge[current_resname]
+
+    @staticmethod
+    def get_counterion_indices(charge_diff,
+                               old_res,
+                               new_res,
+                               new_positions,
+                               new_topology,
+                               positive_ion_name='NA',
+                               negative_ion_name='CL',
+                               radius=0.3):
+        """
+        return a randomized counterion index to neutralize (index w.r.t. new_topology)
+
+        Parameters
+        ----------
+        charge_diff : int
+            the charge difference between the new_system - old_system
+        old_res : openmm.Topology.residue
+            old mutating residue
+        new_res : openmm.Topology.residue
+            new mutating residue
+        new_positions : np.ndarray(N, 3)
+            positions of atoms corresponding to new_topology
+        new_topology : openmm.Topology
+            topology of new system (with new_res)
+        positive_ion_name : str, default 'NA'
+            name of positive ions to identify as candidates for neutralization (mdtraj naming convention)
+        negative_ion_name : str, default 'CL'
+            name of negative ions to identify as candidates for neutralization (mdtraj naming convention)
+        radius : float, default 0.3
```

Comment:
Can we make this a unit-bearing `Quantity`?

Reply:
It should not be unit bearing, as it will be fed into mdtraj.compute_neighbors, which takes in an int.

Reply:
Our APIs should strive to standardize how unit-bearing quantities are passed. We should use unit-bearing quantities, check to make sure the dimensionality is correct internally, and convert internally into whatever the tools we use under the hood (e.g. MDTraj, which uses nm).

@mikemhenry Maybe you can help us clean up some of these API issues?

Reply:
Tracking that here:
https://github.com/choderalab/perses/issues/790